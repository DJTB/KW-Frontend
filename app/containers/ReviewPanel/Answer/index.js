import React from 'react';
// import PropTypes from 'prop-types';
// import styled from 'styled-components';

ReviewAnswer.propTypes = {

};

function ReviewAnswer() {
  return (
    <div>
    </div>
  );
}

export default ReviewAnswer;
