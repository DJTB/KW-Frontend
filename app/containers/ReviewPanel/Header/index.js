import React from 'react';
// import PropTypes from 'prop-types';
// import styled from 'styled-components';

ReviewHeader.propTypes = {

};

function ReviewHeader() {
  return (
    <div>
    </div>
  );
}

export default ReviewHeader;
