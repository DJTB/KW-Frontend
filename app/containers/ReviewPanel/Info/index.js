import React from 'react';
import PropTypes from 'prop-types';

import ToggleBar from './ToggleBar';
import InfoPanel from './InfoPanel';
import NotesPanel from './NotesPanel';
import AddSynonymPanel from './AddSynonymPanel';

const PANELS = ['INFO', 'NOTES', 'ADDSYNONYM'];

export class ReviewInfo extends React.PureComponent {
  static propTypes = {
    reviewEntry: PropTypes.object.isRequired,
    detailLevel: PropTypes.number.isRequired,
    panelToShow: PropTypes.oneOf(Object.values(PANELS)).isRequired,
  }

  render() {
    const { reviewEntry, panelToShow, detailLevel } = this.props;

    const renderNotes = () => <NotesPanel reviewEntry={reviewEntry} />;
    const renderInfo = () => <InfoPanel reviewEntry={reviewEntry} detailLevel={detailLevel} />;
    const renderAddSynonym = () => <AddSynonymPanel addPadding={detailLevel !== 'LOW'} />;

    return (
      <div>
        <ToggleBar />
        {panelToShow === PANELS.NOTES && renderNotes()}
        {panelToShow === PANELS.INFO && renderInfo()}
        {panelToShow === PANELS.ADDSYNONYM && renderAddSynonym()}
      </div>
    );
  }
}

// const mapStateToProps = createStructuredSelector({
//   detailLevel: selectDetailLevel,
//   panelToShow: createSelector([selectPanels], (panels) => panels.show),
// });

export default /* connect(mapStateToProps)*/(ReviewInfo);
