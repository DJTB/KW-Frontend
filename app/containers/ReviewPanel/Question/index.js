import React from 'react';
import PropTypes from 'prop-types';

import {
  Wrapper,
  QuestionWrapper,
  Question,
  Tags,
} from './styles';

ReviewQuestion.propTypes = {
  meanings: PropTypes.array.isRequired,
  tags: PropTypes.array.isRequired,
};

function ReviewQuestion({ meanings, tags }) {
  return (
    <Wrapper>
      <QuestionWrapper>
        <Question>{meanings.join(', ')}</Question>
      </QuestionWrapper>
      <Tags tags={tags} />
    </Wrapper>
  );
}

export default ReviewQuestion;
