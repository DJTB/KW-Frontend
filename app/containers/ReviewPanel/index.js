import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';

import ReviewAnswer from 'containers/ReviewAnswer';
import ReviewHeader from './Header';
import ReviewQuestion from './Question';
import ReviewInfo from './Info';
import { Wrapper, Upper, Lower, ReviewBackground } from './styles';

ReviewPanel.propTypes = {
  reviewEntry: PropTypes.object.isRequired,
};

function ReviewPanel({ reviewEntry }) {
  return (
    <Wrapper>
      <Helmet>
        <title>Review Session</title>
        <meta name="description" content="Kaniwani Review Session" />
      </Helmet>
      <Upper>
        <ReviewHeader
          percentComplete={25}
          percentCorrect={33}
          reviewsComplete={5}
          reviewsRemaining={20}
        />
        <ReviewQuestion
          meanings={['confinement', 'imprisonment', 'incarceration', 'confinement', 'imprisonment', 'incarceration', 'confinement', 'imprisonment', 'incarceration']}
          tags={['Noun', 'Verb', 'Noun', 'Verb', 'Noun', 'Verb', 'Common', 'JLPT N1']}
        />
      </Upper>
      <Lower>
        <ReviewAnswer streak="APPRENTICE" />
        <ReviewInfo reviewEntry={reviewEntry} isDisabled={false} />
        <ReviewBackground />
      </Lower>
    </Wrapper>
  );
}

export default ReviewPanel;
