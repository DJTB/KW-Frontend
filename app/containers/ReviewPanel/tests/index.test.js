import React from 'react';
import { shallow } from 'enzyme';
import ReviewPanel from '../index';

describe('<ReviewPanel />', () => {
  it('should match baseline snapshot', () => {
    expect(shallow(<ReviewPanel />)).toMatchSnapshot();
  });
});
