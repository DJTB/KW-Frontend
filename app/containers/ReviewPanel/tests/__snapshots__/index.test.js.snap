// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewPanel /> should match baseline snapshot 1`] = `
<styled.div>
  <HelmetWrapper
    encodeSpecialCharacters={true}
  >
    <title>
      Review Session
    </title>
    <meta
      content="Kaniwani Review Session"
      name="description"
    />
  </HelmetWrapper>
  <styled.section>
    <ReviewHeader
      percentComplete={25}
      percentCorrect={33}
      reviewsComplete={5}
      reviewsRemaining={20}
    />
    <ReviewQuestion
      meanings={
        Array [
          "confinement",
          "imprisonment",
          "incarceration",
          "confinement",
          "imprisonment",
          "incarceration",
          "confinement",
          "imprisonment",
          "incarceration",
        ]
      }
      tags={
        Array [
          "Noun",
          "Verb",
          "Noun",
          "Verb",
          "Noun",
          "Verb",
          "Common",
          "JLPT N1",
        ]
      }
    />
  </styled.section>
  <styled.section>
    <ReviewAnswer
      answer={
        Object {
          "correct": false,
          "incorrect": false,
          "marked": false,
          "text": "",
          "type": "",
          "valid": true,
        }
      }
      streak="APPRENTICE"
    />
    <ReviewInfo
      addSynonym={
        Object {
          "isActive": false,
        }
      }
      detailLevel="LOW"
      info={
        Object {
          "isActive": false,
        }
      }
      isDisabled={false}
      notes={
        Object {
          "isActive": false,
        }
      }
      reviewEntry={
        Object {
          "id": 12,
          "vocabulary": Object {
            "id": 553,
            "meanings": Array [
              "before long",
              "soon",
            ],
            "readings": Array [
              Object {
                "character": "近々",
                "common": true,
                "id": 582,
                "jlpt": "JLPT N2",
                "kana": "ちかぢか",
                "level": 7,
                "sentenceEn": "When I saw them on a date, I asked them, “Do I hear wedding bells?”",
                "sentenceJa": "彼らがデートをしているのを見かけて近々結婚するのってきいたんだ",
                "tags": Array [
                  "Noun",
                ],
              },
              Object {
                "character": "近々",
                "common": true,
                "id": 583,
                "jlpt": "JLPT N2",
                "kana": "きんきん",
                "level": 7,
                "sentenceEn": "When I saw them on a date, I asked them, “Do I hear wedding bells?”",
                "sentenceJa": "彼らがデートをしているのを見かけて近々結婚するのってきいたんだ",
                "tags": Array [
                  "Noun",
                ],
              },
              Object {
                "character": "近々",
                "common": true,
                "id": 584,
                "jlpt": "JLPT N2",
                "kana": "ちかじか",
                "level": 7,
                "sentenceEn": "When I saw them on a date, I asked them, “Do I hear wedding bells?”",
                "sentenceJa": "彼らがデートをしているのを見かけて近々結婚するのってきいたんだ",
                "tags": Array [
                  "Noun",
                ],
              },
            ],
          },
        }
      }
    />
    <styled.div />
  </styled.section>
</styled.div>
`;
